### PR TITLE
Added Documents for kubernetes.io/enforce-mountable-secrets.

### PR DESCRIPTION
**Pull request by Shubham82**
Added the document for kubernetes.io/enforce-mountable-secrets annotation in [Well-Known Labels, Annotations and Taints](https://kubernetes.io/docs/reference/labels-annotations-taints/).

Fixes: #31853

### CODE REVIEW

**Comment by Shubham82** [comment]
Hi @onlydole @rajeshdeshpande02
Could you please take a look!

**Comment by sftim** [comment]
@kubernetes/sig-auth-pr-reviews how does this look to you? Anything you'd want to add?

File: content/en/docs/reference/labels-annotations-taints.md [hunk 1]
```diff
@@ -80,6 +80,14 @@ Used on: All Objects
 
 This annotation is used for describing specific behaviour of given object.
 
+## kubernetes.io/enforce-mountable-secrets {#enforce-mountable-secrets}
+
+Example: `kubernetes.io/enforce-mountable-secrets: "true"`
+
+Used on: ServiceAccount
+
+The value for this annotation must be **true** to take effect. This annotation indicates that a ServiceAccount should enforce mountable secrets. This field should not be used to find auto-generated service account token secrets for use outside of pods.
```
Comment (liggitt) [inline content/en/docs/reference/labels-annotations-taints.md:89]:
```suggestion
The value for this annotation must be **true** to take effect.
This annotation indicates that pods running as this service account may only reference
Secret API objects specified in the service account's `secrets` field.
```

Comment (Shubham82) [reply content/en/docs/reference/labels-annotations-taints.md:89]: Thanks, @liggitt for taking look at this.
this seems good, I will change it according to your suggestion.

Comment (Shubham82) [reply content/en/docs/reference/labels-annotations-taints.md]: Hi @liggitt
I have addressed the comment, PTAL!

Thanks!

File: content/en/docs/reference/labels-annotations-taints.md [hunk 2]
```diff
@@ -80,6 +80,14 @@ Used on: All Objects
 
 This annotation is used for describing specific behaviour of given object.
 
+## kubernetes.io/enforce-mountable-secrets {#enforce-mountable-secrets}
+
+Example: `kubernetes.io/enforce-mountable-secrets: "true"`
```
Comment (kbhawkey) [inline content/en/docs/reference/labels-annotations-taints.md:85]: nit: Should the value use double quotes? Instead:
Example: `kubernetes.io/enforce-mountable-secrets: true`

Comment (tengqm) [reply content/en/docs/reference/labels-annotations-taints.md:85]: Maybe, maybe not. The keys and values of annotations must be strings. If the double quotes are omitted, k8s may treat this as a boolean? Not sure.

Comment (liggitt) [reply content/en/docs/reference/labels-annotations-taints.md:85]: value should be double quoted as a string

Comment (kbhawkey) [reply content/en/docs/reference/labels-annotations-taints.md:85]: 👍

**Pull request merged**